### DISCUSSION TITLE
Return "true" if a NTE after OBX

### DISCUSSION THREAD

**Discussion by hartlin** [nextgenhealthcare/connect#5633] [General]
Hello All:
Using Mirth Connect ver 4.11

OBX|2|ST|QOQHDL^HDL Cholesterol^L|1|1.10||>0.99|||A^S|F|||202301061738|PPP^Pembroke Regional Hospital^L
OBX|3|ST|QOQTRIG^Triglycerides^L|1|See Note||<1.71|||A^S|F|||202301061738|PPP^Pembroke Regional Hospital^L
NTE|3||No result due to Bilirubin interference, suggest

I need code that will give me a "true" if there is a NTE after an OBX in an If statement.

I have tried using  getAfterSegments in different ways without success.

any help would be appreciated.

**Comment by pacmano1** [comment]
Post your code attempt.

You should be able to call getSegmentsAfter and check length.

**Reply by hartlin** [reply]
agreed, how would you formulate the getSegmentsAfter ?

if ( nte in getSegmentsAfter(obx,'NTE') = true)  would something like this work?

**Reply by pacmano1** [reply]
```JS
for each (obx in msg.OBX) {
 logger.info(getSegmentsAfter(msg, obx , 'NTE', true).length)
}
```
You might want to stop collecting segments if the OBX NTE repeat, please see the docs on getSegmentsAfter

**Reply by pacmano1** [reply]
Please also look over on the forums where many examples of getSegmentsAfter are posted.

Here is a  more complicated one: https://forums.mirthproject.io/forum/mirth-connect/support/174992-getsegmentsafter-for-multiple-obr-segments?p=174997#post174997
